### PR TITLE
Re-harmonize argument parsing between distutils and optparse CLIs

### LINKED ISSUE #388: 2.3.3 distutils extract_messages misparses --keywords
We run in OpenStack to extract only the "_LW" keyword the following:

 python setup.py extract_messages --no-default-keywords \
            --keyword _LW \
            --add-comments Translators: \
            --input-dirs ${modulename} \
            --output-file ${POT}

With babel 2.3.3 the --no-default-keywords has no effect, the files contain far too much content.

### LINKED ISSUE #390: pybabel extract forgot the -s CLI flag in version 2.3
# Overview Description
The -s flag for pybable extract was a shorthand form of --strip-comments. As part of #311 the shorthand form was removed.

# Steps to Reproduce

1. Try to run `pybabel extract -s`

# Actual Results

    $ pybabel extract -s
    Usage: pybabel extract [options] <input-paths>
    
    pybabel: error: no such option: -s

# Expected Results

    $ pybabel extract -s
    Usage: pybabel extract [options] dir1 <dir2> ...
    
    pybabel: error: incorrect number of arguments

# Reproducibility

Introduced with v2.3 with the consolidation of the CLI interfaces.

### PR DESCRIPTION
**Pull request by akx**
I so hope this will be the last we see of this series of bugs...

Fixes #388
Fixes #390
Refs #384
Refs #311

### CODE REVIEW

**Comment by akx** [comment]
Assigning @jtwang just to have some eyes on this.

**Comment by ENuge** [comment]
(Pretend I'm jtwang), shipit! I'm not super familiar with the different ways users can do option parsing, but the logic looks fine and you've got it tested. Hopefully this doesn't introduce any more regressions. :)

**Comment by akx** [comment]
Well, now that #390 appeared, I'm going to add a take a super duper close look at every single parameter that may have been mutated in #311. So re-review after that's done . . .

**Comment by akx** [comment]
Ping @ENuge / @jtwang again.

File: babel/messages/frontend.py [hunk 1]
```diff
@@ -39,6 +39,48 @@
     from configparser import RawConfigParser
 
 
+def listify_value(arg, split=None):
```
Comment (jtwang) [inline babel/messages/frontend.py]: Will a single arg ever contain quotes/whitespace? Eg "foo \"i am a.zip\" bar"

Comment (akx) [inline babel/messages/frontend.py]: That sort of usage will never have been (how's that for a temporal form?) supported by Babel... At least not the optparse frontend.

**Comment by akx** [comment]
@ENuge: Mind taking one more look?

File: babel/messages/frontend.py [hunk 2]
```diff
@@ -825,6 +882,7 @@ def _configure_command(self, cmdname, argv):
             strs = ["--%s" % name]
             if short:
                 strs.append("-%s" % short)
+            strs.extend(cmdclass.option_aliases.get(name, ()))
```
Comment (ENuge) [inline babel/messages/frontend.py]: Checking if this, combined with `strs = ["--%s" % name]`, doesn't cause any problems? If `name` is "output" you'll end up with strs being ['--output', --output']. I don't have a super solid understanding of all the different options uh, options, so for all I know this intentional. If not an issue, shipit! Everything else looks good to me.

Comment (akx) [inline babel/messages/frontend.py]: Oop, good catch.

The alias mapping should be `'output_file': ('--output',)`...

Comment (akx) [inline babel/messages/frontend.py]: (and as of ee8abd6, it is.)

**Comment by akx** [comment]
So yeah, with that last comment fixed, merging. Will roll another bugfix release tomorrow.

File: babel/messages/frontend.py (hunk 2 above)
Comment (ENuge) [inline babel/messages/frontend.py]: Err, oh. I don't know why I was looking at the old version. Shipit!

**Pull request merged**